letsencrypt/models/base_config_settings.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+from inspect import getmembers, isfunction
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    letsencrypt_dns_provider = fields.Selection(
+        selection='_get_supported_dns_providers')
+    letsencrypt_dns_login = fields.Char()
```

Comment:
I think those should be namespaced with the provider, as in `letsencrypt_$providername_$whatever` - it will very often be a login and something like a password, but let's separate theinput of each provider, that's simpler for everyone in the long run

Reply:
Aren't we going to waste a tremendous amount of resources that way? What if this module grows to hold 50 providers. Does that mean that we will have 49 * 3 wasted/empty fields?

Reply:
if a user thinks she must install 50 provider modules (I think in the end every provider should be its own module, probably in some localization repo, just for simplicity I think it's okay to have the first provider here), yes. An empty column costs postgres nearly nothing. Further note that those are virtual fields anyways.